scripts/release.sh

```diff
@@ -13,64 +13,70 @@ if [[ $# != 2 ]] && [[ $# != 3 ]]; then
 fi
 
 if ! [[ -e "$DSS_HOME/application_secrets.json" ]]; then
-    http --check-status "https://$API_DOMAIN_NAME/internal/application_secrets" > "$DSS_HOME/application_secrets.json" || (
+    http --check-status GET "https://$API_DOMAIN_NAME/internal/application_secrets" > "$DSS_HOME/application_secrets.json" || (
         echo "Failed to fetch application_secrets.json. Please create this file and try again."
         exit 1
     )
 fi
 
 if ! diff <(pip freeze) <(tail -n +2 "$DSS_HOME/requirements-dev.txt"); then
-    if [[ $# == 3 ]] && [[ $3 == "--force" ]]; then
+    if [[ "force" == ${DSS_FORCE_RELEASE} ]]; then
         echo "Your installed Python packages differ from requirements-dev.txt. Forcing deployment anyway."
     else
         echo "Your installed Python packages differ from requirements-dev.txt. Please update your virtualenv."
-        echo "Run $0 with --force to deploy anyway."
+        echo "Run $0 with DSS_FORCE_RELEASE='force' force to deploy anyway."
         exit 1
     fi
 fi
 
 export PROMOTE_FROM_BRANCH=$1 PROMOTE_DEST_BRANCH=$2
 
-GH_API=https://api.github.com
-REPO=$(git remote get-url origin | perl -ne '/github\.com.(.+?)(\.git)?$/; print $1')
-STATUS=$(http ${GH_API}/repos/${REPO}/commits/${PROMOTE_FROM_BRANCH}/status Accept:application/vnd.github.full+json)
-STATE=$(echo "$STATUS" | jq -r .state)
-echo "$STATUS" | jq '.statuses[]|select(.state != "success")'
-
-# TODO: (akislyuk) some CI builds no longer deploy or run a subset of tests. Find the last build that ran a deployment.
-if [[ "$STATE" != success ]]; then
-    if [[ $# == 3 ]] && [[ $3 == "--force" ]]; then
-        echo "Status checks failed on branch $PROMOTE_FROM_BRANCH. Forcing promotion and deployment anyway."
-    else
-        echo "Status checks failed on branch $PROMOTE_FROM_BRANCH."
-        echo "Run with --force to promote $PROMOTE_FROM_BRANCH to $PROMOTE_DEST_BRANCH and deploy anyway."
-        exit 1
+if [[ "hca_cicd" != $(whoami) ]]; then
+    # Skip when this script is executed by the GitLab runner
+    GH_API=https://api.github.com
+    REPO=$(git remote get-url origin | perl -ne '/github\.com.(.+?)(\.git)?$/; print $1')
+    STATUS=$(http GET ${GH_API}/repos/${REPO}/commits/${PROMOTE_FROM_BRANCH}/status Accept:application/vnd.github.full+json)
+    STATE=$(echo "$STATUS" | jq -r .state)
+    echo "$STATUS" | jq '.statuses[]|select(.state != "success")'
+    
+    # TODO: (akislyuk) some CI builds no longer deploy or run a subset of tests. Find the last build that ran a deployment.
+    if [[ "$STATE" != success ]]; then
+        if [[ "force" == ${DSS_FORCE_RELEASE} ]]; then
+            echo "Status checks failed on branch $PROMOTE_FROM_BRANCH. Forcing promotion and deployment anyway."
+        else
+            echo "Status checks failed on branch $PROMOTE_FROM_BRANCH."
+            echo "Run with DSS_FORCE_RELEASE='force' to promote $PROMOTE_FROM_BRANCH to $PROMOTE_DEST_BRANCH and deploy anyway."
+            exit 1
+        fi
```

Comment:
Why are status checks skipped here on the CD runner?

Reply:
In order to reach this stage in the GitLab pipeline, testes have necessarily passed on the build box. Do we also wish to require  that Travis tests have passed?

Reply:
I guess you're right, we don't want to depend on travis here.